opendevin/runtime/runtime.py

```diff
@@ -71,6 +81,9 @@ def __init__(
         self.event_stream.subscribe(EventStreamSubscriber.RUNTIME, self.on_event)
         self._bg_task = asyncio.create_task(self._start_background_observation_loop())
 
+    def set_workspace_subdir(self, workspace_subdir: str):
```

Comment:
I'll admit it's not the _best_ practice to change the state on an otherwise fixed-state object, so I'm happy to hear your suggestions.